ayaneo-platform.c

```diff
@@ -927,11 +966,391 @@ struct led_classdev_mc ayaneo_led_mc = {
         .subled_info = ayaneo_led_mc_subled_info,
 };
 
+/* Handling bypass charge */
+struct ayaneo_ps_priv {
+    u8 charge_types;
+    u8 current_start_threshold;
+    u8 current_end_threshold;
+    u8 bypass_available;
+};
+static struct ayaneo_ps_priv ps_priv = { POWER_SUPPLY_CHARGE_TYPE_CUSTOM, 85, 85, 0 };
+
+/* Set the threshold to stop charging the battery above and start charging the battery below */
+static int set_threshold(enum power_supply_property psp, int val)
+{
+    int ret = 0;
+    if (val < 0 || val > 100) {
+        ret = -EINVAL;
+    } else if(psp == POWER_SUPPLY_PROP_CHARGE_CONTROL_END_THRESHOLD) {
+        if (val < ps_priv.current_start_threshold) {
+            ret = -EINVAL;
+        } else {
+            ps_priv.current_end_threshold = val;
+        }
+    } else {
+        if (val > ps_priv.current_end_threshold ) {
+            ret = -EINVAL;
+        } else {
+            ps_priv.current_start_threshold = val;
+        }
+    }
+    return ret;
+}
+/* Activates bypass charge or use normal charging */
+static int set_charge_types(int val)
+{
+    int ret = 0;
+    if ((val == POWER_SUPPLY_CHARGE_TYPE_UNKNOWN) || (val == POWER_SUPPLY_CHARGE_TYPE_CUSTOM)) {
+        ps_priv.charge_types = val;
+    } else {
+        ret = -EINVAL;
+    }
+    return ret;
+}
+/* Get property values for sysfs */
```

Comment:
Attributes typically have a _show and a _store function for their entry points. We should use this convention.

Reply:
I can name it this way, was just looking at the prototype function pointers.